packages/metrics/src/Metrics.ts

```diff
@@ -30,35 +30,57 @@ const DEFAULT_NAMESPACE = 'default_namespace';
  *   * Context manager to create a one off metric with a different dimension
  *
  * ## Usage
+ * 
+ * ### Functions usage with middleware
+ * 
+ * Using this middleware on your handler function will automatically flush metrics after the function returns or throws an error.
+ * Additionally, you can configure the middleware to easily:
+ * * ensure that at least one metric is emitted before you flush them
+ * * capture a `ColdStart` a metric
+ * * set default dimensions for all your metrics
+ * 
+ * @example
+ * ```typescript
+ * import { Metrics, logMetrics } from '@aws-lambda-powertools/metrics';
+ * import middy from '@middy/core';
+ * 
+ * const metrics = new Metrics({ namespace: 'serverlessAirline', serviceName: 'orders' });
+ * 
+ * const lambdaHandler = async (_event: any, _context: any) => {
+ *   ...
+ * };
+ * 
+ * export const handler = middy(lambdaHandler).use(logMetrics(metrics));
+ * ```
  *
  * ### Object oriented way with decorator
  *
  * If you are used to TypeScript Class usage to encapsulate your Lambda handler you can leverage the [@metrics.logMetrics()](./_aws_lambda_powertools_metrics.Metrics.html#logMetrics) decorator to automatically:
- *   * create cold start metric
+ *   * capture a `ColdStart` metric
  *   * flush buffered metrics
  *   * throw on empty metrics
  *
  * @example
  *
  * ```typescript
  * import { Metrics, MetricUnits } from '@aws-lambda-powertools/metrics';
- * import { Callback, Context } from 'aws-lambda';
+ * import { LambdaInterface } from '@aws-lambda-powertools/commons';
  *
- * const metrics = new Metrics({ namespace:'MyService', serviceName:'withDecorator' });
+ * const metrics = new Metrics({ namespace: 'serverlessAirline', serviceName: 'orders' });
  *
- * export class MyFunctionWithDecorator {
+ * class Lambda implements LambdaInterface {
  *
- *   // FYI: Decorator might not render properly in VSCode mouse over due to https://github.com/microsoft/TypeScript/issues/39371 and might show as *@metrics* instead of `@metrics.logMetrics`
+ *   // FYI: Decorator might not render properly in VSCode mouse over due to https://github.com/microsoft/TypeScript/issues/47679 and might show as *@metrics* instead of `@metrics.logMetrics`
  *
  *   @metrics.logMetrics({ captureColdStartMetric: true, throwOnEmptyMetrics: true })
- *   public handler(_event: any, _context: Context, _callback: Callback<any>): void | Promise<void> {
- *    // ...
- *    metrics.addMetric('test-metric', MetricUnits.Count, 10);
- *    // ...
+ *   public handler(_event: any, _context: any): Promise<void> {
```

Comment:
why do we push of `any` type for Context ? I'd prefer to keep `Context` type

Reply:
I hear you. Let me explain why, but I'm open to put it back.

I removed it here because these were the only examples in the whole repo that have it. Even the published docs don't have it.

Removing it here was a way of standardizing the examples. 

Now, we could go two ways:
- put the context type in all examples and docstrings
- leave as it is, but in the docs put a tip/note about `@types/aws-lambda` and explain it can be installed as `devDependency`

Ultimately, since we don't bundle the dependency, having it here in the IDE tooltip could lead to one of two outcomes:
- you copy/write the content of the tooltip and it doesn't work because you don't have the dep installed
- you might think the dependency is mandatory to use Powertools

What do you think?

Reply:
Make sense. Indeed the copy paste that does not work is really frustrating. deal . approving